src/editor/Gui.cpp

```diff
@@ -586,22 +623,81 @@ void Gui::drawMenuBar() {
 				if (map)
 				{
 					logf("Export wad: %s%s%s\n", g_settings.gamedir.c_str(), g_settings.workingdir.c_str(), (map->name + ".wad").c_str());
-					ExportWad(map);
-					logf("Remove all embedded textures\n");
-					map->delete_embedded_textures();
-					if (map->ents.size())
+					if (ExportWad(map))
 					{
-						std::string wadstr = map->ents[0]->keyvalues["wad"];
-						if (wadstr.find(map->name + ".wad" + ";") == std::string::npos)
+						logf("Remove all embedded textures\n");
+						map->delete_embedded_textures();
+						if (map->ents.size())
+						{
+							std::string wadstr = map->ents[0]->keyvalues["wad"];
+							if (wadstr.find(map->name + ".wad" + ";") == std::string::npos)
+							{
+								map->ents[0]->keyvalues["wad"] += map->name + ".wad" + ";";
+							}
+						}
+					}
+				}
+			}
+
+			if (ImGui::MenuItem("Wavefront (.obj)", NULL)) {
+				Bsp* map = app->getMapContainingCamera()->map;
+				if (map)
+				{
+					map->ExportToObjWIP(g_settings.gamedir + g_settings.workingdir);
+				}
+			}
+
+			if (ImGui::BeginMenu("Models")) {
+				for (int r = 0; r < app->mapRenderers.size(); r++)
+				{
+					if (!app->mapRenderers[r]->map)
+						continue;
+					if (ImGui::BeginMenu((std::string("Map ") + app->mapRenderers[r]->map->name + ".bsp").c_str())) {
+						Bsp* map = app->mapRenderers[r]->map;
+						if (map)
 						{
-							map->ents[0]->keyvalues["wad"] += map->name + ".wad" + ";";
+							for (int i = 0; i < map->modelCount; i++)

```

Comment:
Can you add this to the context menu for entities instead? It would be a lot easier to select a model to export in the 3D view than to lookup model numbers.

![image](https://user-images.githubusercontent.com/12087544/110579709-300a9780-811c-11eb-9b38-293283ceacf2.png)
Would fit best below `Transform` I guess.